src/services/tier/useTier.ts

```diff
@@ -0,0 +1,63 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+
+export const TierNames = {
+  BASIC: 'basic',
+  LITE: 'lite',
+  PRO: 'pro',
+  ENTERPRISE: 'enterprise',
+} as const
+
+export const TierSchema = z
+  .object({
+    owner: z.object({
+      plan: z
+        .object({
+          tierName: z.nativeEnum(TierNames),
+        })
+        .nullish(),
+    }),
```

Comment:
The `owner` field here could also be `null` so needs to have `nullish()` added on, highly recommend pasting your query into playground and checking the types there, if it doesn't have a `!`, that indicates that the value could be returned as `null`.